### ISSUE TITLE
S1854 false positive on array slice

### ISSUE THREAD

**Issue by ian-hamlin** [SonarSource/sonar-dotnet#3912]
### Description

S1854 is warning of an unused variable when using an array slice.

### Repro steps

```
        /// <summary>
        /// Sample showing the array slice false positive.
        /// </summary>
        public void SampleArraySlice()
        {
            var data = new byte[] { 1, 2, 3, 4, 5, 6, 7 };
            var startIndex = new System.Index(2);
            var endIndex = new System.Index(5);
            var slice = data[startIndex..endIndex];
            System.Diagnostics.Contracts.Contract.Assert(slice.Length == 3);
        }
```

### Expected behavior

No rule is triggered.

### Actual behavior

Remove this useless assignment to local variable 'startIndex'.
Remove this useless assignment to local variable 'endIndex'.

### Known workarounds

Ignore the problem in the UI.

### Related information
* sonar-scanner-msbuild-5.0.4.24009-net5.0
* Visual Studio 2019 (16.8.3)
* dotnet core 3.1
* Windows 10
* SonarScanner for MSBuild 5.0.4
* Using the .NET Core version of the Scanner for MSBuild

**Comment by pavel-mikula-sonarsource** [comment]
Hi @ian-hamlin ,

Thank you for reporting this case. I can confirm it as False Positive. Indices and ranges are not supported by this rule. We already have this problem reported in #3348 that you can track. I'll close this one.
